### ISSUE TITLE
Would it be possible to have a direct link between the shape tree and its description? Here it seems that we have a link between a document that includes multiple shape tree definitions to descriptions which feels like weird indirection.

### ISSUE THREAD

**Issue by justinwb** [shapetrees/specification#72]
Would it be possible to have a direct link between the shape tree and its description? Here it seems that we have a link between a document that includes multiple shape tree definitions to descriptions which feels like weird indirection.

Response from @justinwb:

> Note that the shape tree description links directly to the shape tree via st:describes. The benefit of having the descriptions link to the shape vs. the other way around, is that we can expand those more freely without having to directly modify the composition of the shape trees definitions themselves.

_Originally posted by @elf-pavlik in https://github.com/shapetrees/specification/pull/71#discussion_r742314018_

**Comment by srosset81** [comment]
I agree with @elf-pavlik that the current spec make it *really hard* to simply find a description for a given shapetree.

For example if I want to find the english label for a `pm:Project` shapetree:

- First, I need to go through all `st:Description` resources, wherever they may be, and find the ones that are related to the `pm:Project` shapetree (via the `st:describes` predicate). I will get one `st:Description` per available language.
- Then, I need to load all `st:DescriptionSet` related to these `st:Description` (via the `st:inDescriptionSet` predicate) until I find the one related to the english language.

Depending on the way data are stored, this may mean going through dozens of resources... just to get a single description! IMO this breaks the whole interest of linked data, where you can find more information about something just by following related links. 

Imagine if the whole web worked with inverse links like you do here! :smile: 

> Note that the shape tree description links directly to the shape tree via st:describes. The benefit of having the descriptions link to the shape vs. the other way around, is that we can expand those more freely without having to directly modify the composition of the shape trees definitions themselves.

What is wrong with modifying the shapetree definition if a new language is added ?

**Comment by elf-pavlik** [comment]
> Imagine if the whole web worked with inverse links like you do here! 😄

My approach is to always have backlinks for those inverse links.  Since directions of predicates are pretty arbitrary, I treat subjects and objects equally. 

For example https://github.com/janeirodigital/sai-js/blob/main/packages/css-storage-fixture/dev/shapetrees/trees/Project%24.ttl#L32-L51

Since shape definitions support inverse relations https://shex.io/shex-primer/#inverse-properties
I want to use it to specify when backlinks have to be included.

**Comment by srosset81** [comment]
> For example https://github.com/janeirodigital/sai-js/blob/main/packages/css-storage-fixture/dev/shapetrees/trees/Project%24.ttl#L32-L51

So, if I understand well, you copy these lines...

```ttl
solidtrees:desc-en\#Project
  shapetrees:inDescriptionSet solidtrees:desc-en ;
  shapetrees:describes solidtrees:Project .

solidtrees:desc-en
  shapetrees:usesLanguage "en"^^xsd:language .
```

... from [this file](https://github.com/janeirodigital/sai-js/blob/main/packages/css-storage-fixture/dev/shapetrees/trees/desc-en%24.ttl) to [this file](https://github.com/janeirodigital/sai-js/blob/main/packages/css-storage-fixture/dev/shapetrees/trees/Project%24.ttl) ?

If so, this means you need to keep a synchronization of data between files. Which for me is a bad practice, because there should only be a single source of truth. From my perspective, the whole point of linked data is to avoid making copies of the same data ! (otherwise why would we bother with URIs ?)
